src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -80,7 +84,9 @@ export class RelayFeeCalculator {
       console.error(`ERROR(gasFeePercent): Error while fetching token price ${error}`);
       throw error;
     });
-    const [gasCosts, tokenPrice] = await Promise.all([getGasCosts, getTokenPrice]);
+    const results = await Promise.all(_tokenPrice !== undefined ? [getGasCosts, getTokenPrice] : [getGasCosts]);
+    const gasCosts = results[0];
+    const tokenPrice = (_tokenPrice !== undefined ? _tokenPrice : results[1]) as number;
```

Comment:
Promise.all works on non-promises, too!

```suggestion
    const [gasCosts, tokenPrice] = await Promise.all([getGasCosts, _tokenPrice !== undefined ? _tokenPrice : getTokenPrice]);
```